### DISCUSSION TITLE
Can't Verify Email Address

### DISCUSSION THREAD

**Discussion by wptracy** [dani-garcia/vaultwarden#3201] [Q&A] [answered]
After creating privkey.pem and fullchain.pem on a different ubuntu server and uploading them to Home Assistant /ssl,
I was able to get a login page to create an account at https://10.0.0.233:7277/#/lock. and login.

openssl genrsa > privkey.pem
openssl req -new -x509 -key privkey.pem > fullchain.pem

Now I can't Verify my account's email address to unlock access to all features. and I can't authorize or deauthorize devices that have access to my account.

I can't administer vaultwarden until I verify my accounts email address and unlock access.

Email links don't work:

- [Verify Email Address Now](https://localhost/#/verify-email/?userId=a5e97cfa-8067-4538-8d49-1a5b53d1c983&token=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJuYmYiOjE2NzUwNjc4MzMsImV4cCI6MTY3NTQ5OTgzMywiaXNzIjoiaHR0cDovL2xvY2FsaG9zdHx2ZXJpZnllbWFpbCIsInN1YiI6ImE1ZTk3Y2ZhLTgwNjctNDUzOC04ZDQ5LTFhNWI1M2Q
- You can deauthorize all devices that have access to your account from the [web vault](http://localhost/) under Settings > My Account > Deauthorize Sessions.

These are the emails I get:
```
Verify this email address for your account by clicking the link below.

- Verify Email Address Now

If you did not request to verify your account, you can safely ignore this email.

- Your account was just logged into from a new device.

Date: Monday, January 30, 2023 at 02:57:47 AM MST

IP Address: 10.0.0.113
Device Type: Firefox
You can deauthorize all devices that have access to your account from the web vault under Settings > My Account > Deauthorize

Your account was just logged into from a new device.
Date: Monday, January 30, 2023 at 02:57:47 AM MST
IP Address: 10.0.0.113
Device Type: Firefox
You can deauthorize all devices that have access to your account from the [web vault](http://localhost/) under Settings > My Account > Deauthorize
```

**Comment by BlackDex** [comment, marked as answer]
You need to set the correct `DOMAIN` variable so that the links will work.

**Reply by wptracy** [reply]
I tried changing localhost to the IP of Home Assistant, and that didn't work. Unable to connect

Is that what you mean by DOMAIN ?

I also tried adding port 8123 and got this:

Secure Connection Failed

An error occurred during a connection to 10.0.0.233:8123. SSL received a record that exceeded the maximum permissible length.

Error code: SSL_ERROR_RX_RECORD_TOO_LONG

    The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.
    Please contact the website owners to inform them of this problem.

**Reply by BlackDex** [reply]
I suggest to check your environment if all is set correctly. We recommend to use a reverse proxy, also so that web-sockets work.
The `DOMAIN` variable is something you need to set as an option/variable for Vaultwarden.

See:
- https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples
- https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview
- https://github.com/dani-garcia/vaultwarden/wiki/Using-Docker-Compose

**Reply by wptracy** [reply]
In Proxy-examples, I read this:

Note that when you put vaultwarden behind a reverse proxy, the connections between the reverse proxy and vaultwarden are typically assumed to be going through a secure private network, and thus do not need to be encrypted. The examples below assume you are running in this configuration, in which case you should not enable the HTTPS functionality built into vaultwarden (i.e., you should not set the ROCKET_TLS environment variable). If you do, connections will fail since the reverse proxy is using HTTP to connect to vaultwarden, but you're configuring vaultwarden to expect HTTPS.

Am I not supposed to be using SSL in the configuration of Home Assistant Vaultwarden (Bitwarden)
See Screenshot below.

And where do I set a DOMAIN.  I copied the https://github.com/dani-garcia/vaultwarden/blob/main/.env.template into .env and rebooted.  When I rebooted it erased the .env file.

I entered my duckdns domain in https://10.0.0.233:7277/admin settings and tried the email verification again.  It tries to connect to xxxxxxx.duckdns.org/#/verify-email/?userId=a5e97cfa-8067-4538-8d49-1a5b53d1c983&token=eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9......
but times out.

I added https://10.0.0.233:7277/ to Domain URL in https://10.0.0.233:7277/admin setting, and now I think everything works.

I then added the Bitwarden extension to firefox and was able to configure my Home Assistant URL in it.

This youtube was helpful. https://www.youtube.com/watch?v=DdCMhchFtCc

```
## Domain settings
## The domain must match the address from where you access the server
## It's recommended to configure this value, otherwise certain functionality might not work,
## like attachment downloads, email links and U2F.
## For U2F to work, the server must use HTTPS, you can use Let's Encrypt for free certs
# DOMAIN=https://vw.domain.tld:8443
```

![Screenshot from 2023-01-30 04-19-49](https://user-images.githubusercontent.com/78648145/215463713-30136123-6701-4e57-b7be-1dc04abee600.png)
